app/models/post.rb

```diff
@@ -1035,7 +1035,7 @@ def link_post_uploads(fragments: nil)
   end
 
   def update_uploads_secure_status(source:)
-    if Discourse.store.external?
+    if Discourse.store.external? && SiteSetting.secure_uploads?
```

Comment:
This is the main change, just having the site setting check here instead of on all the callers is better